modules/install-consul/install-consul

```diff
@@ -182,20 +197,27 @@ function create_consul_install_paths {
   sudo chown -R "$username:$username" "$path"
 }
 
-function install_binaries {
+function fetch_binary {
   local readonly version="$1"
-  local readonly path="$2"
-  local readonly username="$3"
+  local download_url="$2"
+
+  if [[ -z "$download_url" && -n "$version" ]];  then
+    download_url="https://releases.hashicorp.com/consul/${version}/consul_${version}_linux_amd64.zip"
+  fi
 
-  local readonly url="https://releases.hashicorp.com/consul/${version}/consul_${version}_linux_amd64.zip"
-  local readonly download_path="/tmp/consul_${version}_linux_amd64.zip"
-  local readonly bin_dir="$path/bin"
+  log_info "Downloading Consul from $download_url to $DOWNLOAD_PACKAGE_PATH"
+  curl -o "$DOWNLOAD_PACKAGE_PATH" "$download_url"
```

Comment:
Unrelated to this PR, but we now add `--location --silent --fail --show-error` to all `curl` commands to follow redirects, hide useless output, use a non-zero exit code on errors, and show the error messages.